core/src/main/scala/com/raphtory/api/analysis/table/TableOutput.scala

```diff
@@ -5,52 +5,63 @@ import com.raphtory.api.progresstracker._
 import com.raphtory.api.time.Perspective
 import com.typesafe.config.Config
 
+import scala.collection.immutable.SeqMap
+import scala.collection.immutable.SortedSet
+
 /** Concrete Table with computed results for a perspective */
 case class TableOutput private (
     jobID: String,
     perspective: Perspective,
+    header: List[String],
     rows: Array[Row],
     private val conf: Config
 ) extends TableBase {
 
-  /** Add a filter operation to table
-    *
-    * @param f function that runs once for each row (only rows for which `f ` returns `true` are kept)
-    */
+  /** Returns an array of rows represented as arrays of values */
+  def rowsAsArrays(): Array[Array[Any]] = rows.map(row => row.columns.values.toArray)
+
   override def filter(f: Row => Boolean): TableOutput = copy(rows = rows.filter(f))
 
-  /** Explode table rows
-    *
-    * This creates a new table where each row in the old table
-    * is mapped to multiple rows in the new table.
-    *
-    * @param f function that runs once for each row of the table and maps it to new rows
-    */
-  override def explode(f: Row => IterableOnce[Row]): TableOutput = copy(rows = rows.flatMap(f))
+  override def explode(columns: String*): TableOutput = {
+    val explodedRows = rows.flatMap { row =>
+      val validColumns = columns.filter(col => row.get(col) != "")
+      validColumns.map(col => row.get(col).asInstanceOf[Iterable[Any]]).transpose.map { values =>
+        validColumns.zip(values).foldLeft(row) { case (row, (key, value)) => new Row(row.columns.updated(key, value)) }
+      }
+    }
+    this.copy(rows = explodedRows)
+  }
+
+  override def renameColumns(columns: (String, String)*): TableOutput = {
+    val newNames      = columns.toMap
+    val renamedHeader = header.collect {
+      case key if newNames contains key => newNames(key)
+      case key                          => key
+    }
+    val renamedRows   = rows
+      .map { row =>
+        val renamedColumns = row.columns.collect {
+          case (key, value) if newNames contains key => (newNames(key), value)
+          case tuple                                 => tuple
+        }
+        new Row(renamedColumns)
+      }
+    this.copy(rows = renamedRows, header = renamedHeader)
+  }
 
-  /** Write out data and
```

Comment:
Why were these comments removed

Reply:
these comments are in the superclass so no need to duplicate the docs